orc8r/cloud/go/obsidian/reverse_proxy/reverse_proxy.go

```diff
@@ -21,22 +22,49 @@ import (
 	"github.com/labstack/echo/middleware"
 )
 
+// ReverseProxyHandler tracks registered paths to their associated proxy
+// backends. This is used to dynamically update the proxy middleware based
+// off of the service registry
+type ReverseProxyHandler struct {
+	proxyBackendsByPathPrefix map[string]*reverseProxyBackend
+}
+
+type reverseProxyBackend struct {
+	serverUrl *url.URL
+	active    bool
+}
+
+// NewReverseProxyHandler initializes a ReverseProxyHandler
+func NewReverseProxyHandler() *ReverseProxyHandler {
+	return &ReverseProxyHandler{
+		proxyBackendsByPathPrefix: map[string]*reverseProxyBackend{},
+	}
+}
+
 // AddReverseProxyPaths adds reverse proxying from the echo server to every
 // service that has registered obsidian handlers. The proxying is based off of
 // the path prefixes that have been registered in the services' annotations.
-func AddReverseProxyPaths(server *echo.Echo) (*echo.Echo, error) {
+func (r *ReverseProxyHandler) AddReverseProxyPaths(server *echo.Echo) (*echo.Echo, error) {
 	pathPrefixesByAddr, err := getEchoServerAddressToPathPrefixes()
```

Comment:
actually let's decouple the querying of the service registry and the population of the reverse proxy. pass in the echo server:path prefix map as an arg and export that function if you're going to have callsites outside of this package. should simplify the test logic a bit too